### PR TITLE
ZK-4208: Can't close a notification

### PR DESCRIPTION
**Pull request by plijyfes**

### CODE REVIEW

File: zul/src/archive/web/js/zul/box/Layout.js [hunk 1]
```diff
@@ -89,9 +89,10 @@ zul.box.Layout = zk.$extends(zk.Widget, {
 	onChildVisible_: function (child) {
 		this.$supers('onChildVisible_', arguments);
 		if (this.desktop) {
+			var n = child.$n('chdex');
 			this._shallSize = true;
 			//Bug ZK-1650: change chdex display style according to child widget
-			child.$n('chdex').style.display = child.isVisible() ? '' : 'none';
+			if (n) n.style.display = child.isVisible() ? '' : 'none';
```
Comment (DevChu) [inline zul/src/archive/web/js/zul/box/Layout.js]: should track root cause

File: zul/src/archive/web/js/zul/wgt/Notification.js [hunk 2]
```diff
@@ -227,6 +227,28 @@ zul.wgt.Notification = zk.$extends(zul.wgt.Popup, {
...
+					zk.afterAnimate(function () {zUtl.fireShown(wgt);}, -1);
+				} else {
+					this.fire('onHide');
+					zWatch.fireDown('onHide', this);
+					this.setDomVisible_(node, false);
+				}
+			}
+			jq.onSyncScroll(this);
+		}
+		return this;
```
Comment (DevChu) [inline zul/src/archive/web/js/zul/wgt/Notification.js]: create a new issue about popup fakeparent

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**